src/app/core/common-components/entity-form/entity-form.service.ts

```diff
@@ -111,15 +122,17 @@ export class EntityFormService {
    * @param forTable
    */
   public createFormGroup<T extends Entity>(
-    formFields: FormFieldConfig[],
+    formFields: ColumnConfig[],
     entity: T,
     forTable = false,
   ): EntityForm<T> {
-    this.extendFormFieldConfig(formFields, entity.getConstructor(), forTable);
+    const fullFields = formFields.map((f) =>
+      this.extendFormFieldConfig(f, entity.getConstructor(), forTable),
+    );
```

Comment:
with the new `<app-form-field` NoteDetails (and others) do not need field details anymore, so that should be fine to not run this in those components. 

To create the form with its validators, we do need the field schema in that service. But currently it's actually a weird mix of loading the schema and using the extended field ...
Doing extendFormFieldConfig here also means it will happen twice - again in the FormFieldComponent. Although that shouldn't create much problems.

Crazy other idea: What if FormFieldComponents would self-register their formControls to the given (empty) form object passed as input?

Reply:
Then lets split it up. The validators are only used to create the form group and not in the components. At the moment I think the validators always come from the schema and never from the field definition which I think also makes sense. 
Regarding the self registration. I am not sure whether its a good idea to move all this logic to the components instead of having it in a central service. Or do you see any advantages?